src/components/UI/CartInit/CartInit.tsx

```diff
@@ -2,25 +2,35 @@
 import { useAuthStore } from '@/store/authStore'
 import { useCombinedStore } from '@/store/store'
 import { useShallow } from 'zustand/react/shallow'
+import { useStoreData } from '@/hooks/useStoreData'
+import { useEffect } from 'react'
 
 export default function CartInit() {
   const itemsIds = useCombinedStore(useShallow((state) => state.itemsIds))
   const getCartItems = useCombinedStore((state) => state.getCartItems)
   const syncBackendCart = useCombinedStore((state) => state.syncBackendCart)
   const isSync = useCombinedStore((state) => state.isSync)
   const reset = useCombinedStore((state) => state.resetCart)
-  const { token } = useAuthStore()
 
-  if (!token) {
-    if (isSync) {
-      reset()
-    }
-    if (itemsIds.length) {
-      getCartItems().catch((e) => console.log(e))
+  const token = useStoreData(
+    useAuthStore,
+    (state) => state.token,
+  )
+
+  useEffect(() => {
```

Comment:
nit: `useEffect` turned out to be useful here :)

Reply:
Yes! It turns out that the value of the counter was passed not from the state, but for some unknown reason from the type! Also, when logging into the account, we had an error in the post request for aitems and had to call the token with a delay to get rid of this error! Now all the dependencies are transferred to the useEffect and everything seems to work properly

Reply:
Cool, thanks for explanation